src/components/charts/SummaryPieChart.jsx

```diff
@@ -0,0 +1,162 @@
+      percentLeft -= percentage;
+      return Object.assign({ percentage }, entry);
+    });
+  }
+  return data;
+};
+
+class SummaryPieChart extends React.Component {
+  render() {
+    const useTwoColors = this.props.data.length === 2;
+    const getCategoryColor = (index) => {
+      if (useTwoColors) {
+        return this.props.localTheme[`pieChartTwoColor.pie${(index % 2) + 1}Color`];
+      }
+      const i = (index % 9) + 1;
+      return this.props.localTheme[`barGraph.bar${i}Color`];
+    };
+    const barChartData = calculateBarChartData(
+      this.props.data,
+      this.props.showPercentage,
+      this.props.percentageFixedPoint,
+    );
+    const dataKey = this.props.showPercentage ? 'percentage' : 'value';
+    const toPercentageFormatter = v => (this.props.showPercentage ? `${v}%` : v);
+    return (
+      <PieChartWrapper>
+        <PieChartTitle className="h4-typo">
+          {this.props.title}
+        </PieChartTitle>
+        <PieChartInnerWrapper>
+          <PieChartLegendWrapper>
+            {
+              barChartData.map(entry => (
+                <LegendWrapper className="form-body" key={'text'.concat(entry.name)}>
+                  <LegendName>
+                    {entry.name}
+                  </LegendName>
+                  <LegendValue className="form-special-number">
+                    {
+                      this.props.showPercentage ? `${entry[dataKey]}%` : entry[dataKey]
+                    }
+                  </LegendValue>
+                </LegendWrapper>))
+            }
+          </PieChartLegendWrapper>
+          <PieChart
+            width={this.props.outerRadius * 2}
+            height={this.props.outerRadius * 2}
+            style={pieChartStyle}
+          >
+            <Pie
+              dataKey={dataKey}
+              isAnimationActive={false}
+              data={barChartData}
+              innerRadius={this.props.innerRadius}
+              outerRadius={this.props.outerRadius}
+              fill="#8884d8"
+            >
+              {
+                barChartData.map((entry, index) => (
+                  <Cell
+                    key={'pie'.concat(index)}
+                    dataKey={dataKey}
+                    fill={getCategoryColor(index)}
+                  />))
+              }
+            </Pie>
+            <Tooltip formatter={toPercentageFormatter} />
+          </PieChart>
+        </PieChartInnerWrapper>
+      </PieChartWrapper>
+    );
+  }
+}
+
+const ChartDataShape = PropTypes.shape({
+  name: PropTypes.string.isRequired,
+  value: PropTypes.number.isRequired,
+});
+
+SummaryPieChart.propTypes = {
+  title: PropTypes.string.isRequired,
+  data: PropTypes.arrayOf(ChartDataShape).isRequired,
+  localTheme: PropTypes.object.isRequired,
+  innerRadius: PropTypes.number,
+  outerRadius: PropTypes.number,
+  showPercentage: PropTypes.bool,
+  percentageFixedPoint: PropTypes.number,
```

Comment:
I'm not sure what this prop does exactly

Reply:
This is for how many digits after point, like "%.2f" in cpp.printf